modules/ROOT/pages/test-ssl-enabled-sources-cookbook.adoc

```diff
@@ -0,0 +1,139 @@
+= Testing SSL-Enabled Applications
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+To test an application that has an HTTP Listener with SSL configured, you must configure an SSL truststore in your test to send authenticated requests to your code using a TLS certificate.
+
+== Example
+
+Assume that you want to test the following Mule application:
+
+[source,xml,linenums]
+----
+<tls:context name="TLS_Context">  //<1>
+  <tls:trust-store path="client_truststore.ts" password="123456" insecure="false" />
+  <tls:key-store type="jks" path="server.jks" alias="serverkey" keyPassword="123456" password="123456"/>
+</tls:context>
+
+<http:listener-config name="HTTP_Listener_config" doc:name="HTTP Listener config" basePath="/">
+  <http:listener-connection host="0.0.0.0" port="8083" protocol="HTTPS" tlsContext="TLS_Context" /> //<2>
+</http:listener-config>
+
+<flow name="tlsExampleFlow">
+  <http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="/"/>
+  <set-payload value='#[output application/json --- {"name": "John Doe"}]' doc:name="Set Payload"/>
+</flow>
+----
+
+Note that:
+
+<1> This application defines a TLS context.
+<2> The HTTP Lister is validating all requests using the TLS context previously defined.
+
+Depending on how you created your test, there are two ways of testing the previous example:
+
+* If you are writing your test, invoke the flow being tested using a `flow-ref` component. +
+See the <<flow-ref-test,Testing Using a Flow-Ref Component>> section for more information.
+* If you scaffolded your test using APIkit, your test uses HTTP requesters to reach the application endpoints. +
+See the <<request-test, Testing Using an HTTP Requester>> section for more information.
+
+[[flow-ref-test]]
+== Testing Using a Flow-Ref Component
+
+To validate the behavior of your flow, your MUnit test must invoke the flow you want to test using a `flow:ref` component. In this case, the Execution scope in your MUnit test must looks as follows:
+
+[source,xml,linenums]
+----
+(...)
+
+<munit:test name="example-using-tls-context" description="Test">
+  (...)
+  <munit:execution >
+    <flow-ref doc:name="Flow Reference" name="tlsExampleFlow"/>
+  </munit:execution>
+(...)
+----
+
+A full test to, for example, assert the payload of the application would look as follows:
+
+[source,xml,linenums]
+----
+	<munit:config name="tls-example-test-suite.xml" />
+
+  <munit:test name="example-using-flow-ref">
+		<munit:execution >
+			<flow-ref doc:name="Flow Reference" name="tlsExampleFlow"/>
+		</munit:execution>
+		<munit:validation >
+			<munit-tools:assert-that doc:name="Assert that" expression="#[payload.name]" is='#[MunitTools::equalTo("John Doe")]'/>
+		</munit:validation>
+	</munit:test>
+
+----
+
```

Comment:
Is there anything useful to add here about the effect that using this method has? What happens as a result?
If not, don't force it!  :-)